Firestore/core/src/model/target_index_matcher.h

```diff
@@ -17,6 +17,7 @@
 #ifndef FIRESTORE_CORE_SRC_MODEL_TARGET_INDEX_MATCHER_H_
 #define FIRESTORE_CORE_SRC_MODEL_TARGET_INDEX_MATCHER_H_
 
+#include <set>
```

Comment:
It is generally preferable to use `unordered_set` instead of `set` because `unordered_set` has better performance. `unordered_set` is a "hash set" where `set` is an "ordered set" implemented by a linked list, and linked lists are virtually always worse peforming due to their CPU cache unfriendliness. Since this set will always be very small (probably no more than 10-20 elements) the performance consideration probably doesn't matter; however, if it's equally easy to use an `unordered_set` then please use it instead. As a benefit, you'll be able to delete that comment explaining that ordering is not required ;)

Reply:
Update: Since the other SDKs will use a sorted set, for consistency, probably just leave the usage of `set` here (rather than changing to `unordered_set`).